scarb/tests/build_targets.rs

```diff
@@ -146,3 +146,56 @@ fn compile_dep_not_a_lib() {
             error: could not compile `hello` due to previous error
         "#});
 }
+
+#[test]
+fn target_with_source_path() {
+    let t = TempDir::new().unwrap();
+    t.child("src/test.cairo")
+        .write_str(r#"fn f() -> felt252 { 42 }"#)
+        .unwrap();
+    t.child("Scarb.toml")
+        .write_str(
+            r#"
+            [package]
+            name = "hello"
+            version = "0.1.0"
+
+            [[target.test]]
+            source-path = "src/test.cairo"
```

Comment:
```suggestion
            source-path = "test/x.cairo"
```

let's try this, to check if this works if we omit `src/` convention. wdyt?